src/components/forms/AddSeparatorForm.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <section class="general-form">
+    <h2>Separadores</h2>
+    <div v-if="updatingShelf">
+      <form @submit.prevent="updateSeparators">
+        <div class="furniture-input">
+          <label>Cuantos Separadores ?</label>
+          <select
+            name="amountOfSeparators"
+            id="amountOfSeparators"
+            v-model.number="amountOfSeparators"
+            @change="updateSeparators"
+          >
+            <option value="0">0</option>
+            <option value="1">1</option>
+            <option value="2">2</option>
+            <option value="3">3</option>
+          </select>
+        </div>
+        <div class="furniture-input">
+          <label>Que orientacion ?</label>
+          <select
+            name="typeOfSeparators"
+            id="typeOfSeparators"
+            v-model="typeOfSeparators"
+            @change="updateSeparators"
+          >
+            <option value="centered">centrado</option>
+            <option value="left">izquierda</option>
+            <option value="right">derecha</option>
+          </select>
+        </div>
+        <button>Confirmar</button>
+      </form>
+    </div>
+    <div v-else>
+      <h2>Haz click en el espacio para agregar separadores</h2>
+    </div>
+  </section>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+export default {
+  props: ['myUpdatingShelf'],
+  data() {
+    return {
+      amountOfSeparators: 0,
+      typeOfSeparators: 'centered',
+    };
+  },
+  computed: {
+    ...mapGetters(['shelfs']),
+    updatingShelf() {
+      return this.shelfs.find((shelf) => shelf === this.myUpdatingShelf);
+    },
+
+    widthVariationRelatedToTypeOfSeparator() {
+      const objectTypeSeparators = {
+        centered: 0,
+        left: -12 / this.amountOfSeparators,
+        right: 12 / this.amountOfSeparators,
+      };
+      return objectTypeSeparators[this.typeOfSeparators];
+    },
+    spaceWidth() {
+      const mapSpaceWidth = new Map([
+        ([0, 100],
+        [1, 50 + this.widthVariationRelatedToTypeOfSeparator],
+        [2, 33.3 + this.widthVariationRelatedToTypeOfSeparator],
+        [3, 25 + this.widthVariationRelatedToTypeOfSeparator]),
+      ]);
+      return mapSpaceWidth.get(this.amountOfSeparators);
+    },
+
+    remainingWidth() {
+      return 100 - this.amountOfSeparators * this.spaceWidth;
+    },
+  },
+  methods: {
+    updateSeparators() {
+      this.updatingShelf.insideSpaces.amountOfSeparators =
+        this.amountOfSeparators;
+      this.updatingShelf.insideSpaces.typeOfSeparators = this.typeOfSeparators;
+      this.pushSpacesIntoArray();
+      this.resetData();
```

Comment:
so here it would be cool to see "updateSeparators()" that will modify the shelf in the local (component) store.
Then at the end, "saveShelfToStore"